src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -769,9 +769,13 @@ public void M(int[] a)
 " + TestSources.GetSubArray;
             var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns);
             compilation.MakeMemberMissing(SpecialMember.System_Array__Length);
```

Comment:
:memo: Note the missing diag on the indexer access `a[^1]` (same when `WellKnownMember.System_Array__get_Length` is marked as missing)